src/cwe_checker_lib/src/analysis/forward_interprocedural_fixpoint.rs

```diff
@@ -283,3 +284,149 @@ pub fn create_computation_with_alternate_worklist_order<'a, T: Context<'a>>(
+            },
+        };
+
+        let caller_block_2 = Term {
+            tid: Tid::new("caller_block_2"),
+            term: Blk {
+                defs: vec![],
+                jmps: vec![],
+                indirect_jmp_targets: Vec::new(),
+            },
+        };
+        let caller_block_1 = Term {
+            tid: Tid::new("caller_block_1"),
+            term: Blk {
+                defs: vec![],
+                jmps: vec![Term {
+                    tid: Tid::new("call"),
+                    term: Jmp::Call {
+                        target: called_function.tid.clone(),
+                        return_: Some(caller_block_2.tid.clone()),
+                    },
+                }],
+                indirect_jmp_targets: Vec::new(),
+            },
+        };
+        let caller_block_2 = Term {
+            tid: Tid::new("caller_block_2"),
+            term: Blk {
+                defs: vec![],
+                jmps: vec![Term {
+                    tid: Tid::new("jmp"),
+                    term: Jmp::Branch(caller_block_1.tid.clone()),
+                }],
+                indirect_jmp_targets: Vec::new(),
+            },
+        };
+        let caller_function = Term {
+            tid: Tid::new("caller_function"),
+            term: Sub {
+                name: "caller_function".to_string(),
+                blocks: vec![caller_block_1, caller_block_2],
+                calling_convention: Some("_stdcall".to_string()),
+            },
+        };
+        let mut project = Project::mock_x64();
+        project
+            .program
+            .term
+            .subs
+            .insert(caller_function.tid.clone(), caller_function.clone());
+
+        project
+            .program
+            .term
+            .subs
+            .insert(called_function.tid.clone(), called_function.clone());
+
+        project
+    }
+
+    #[test]
+    /// Checks if the nodes corresponding to the callee function are first in the worklist.
+    fn check_callee_first_worklist() {
+        let project = mock_project();
+        let extern_subs = project
+            .program
+            .term
+            .extern_symbols
+            .keys()
+            .cloned()
+            .collect();
+
+        let graph = crate::analysis::graph::get_program_cfg(&project.program, extern_subs);
+        let context = Context::new(&graph);
+        let comp = create_computation_with_callee_before_caller_worklist_order(
+            context,
+            Some(HashMap::new()),
+        );
+        // The first two nodes should belong to the callee
+        for node in comp.get_worklist()[0..1].iter() {
+            match graph[*node] {
+                crate::analysis::graph::Node::BlkStart(_, sub)
+                | crate::analysis::graph::Node::BlkEnd(_, sub) => {
+                    assert_eq!(
+                        sub.term,
+                        project.program.term.subs.get(&sub.tid).unwrap().term
```

Comment:
Why not compare TIDs here instead? I know, it is just a unit test, so speed is not important, but it still feels unnecessary to compare the whole functions if comparing their TIDs would  be enough.